### DISCUSSION TITLE
Need all iOS simulators for all Xcode versions.

### DISCUSSION THREAD

**Discussion by jshier** [community/community#25868] [Actions]
<p>At this time the <code>macOS-latest</code> image is missing quite a few combinations of devices and iOS versions, making it very difficult to get good test coverage. It would be best if all versions of Xcode on the macOS VMs should have access to all OS / simulator combinations. For example, a default <code>macOS-latest</code> run, setting the <code>DEVELOPER_DIR</code> to <code>/Applications/Xcode\_11.app/Contents/Developer</code> results in a failure for many OS / device combinations, as the available list is bizarrely limited to:</p>
<p>{ platform:iOS Simulator, id:29611669-E8FD-4824-A5A2-CA2FECA0EE61, OS:10.0, name:iPad Air }<br>
{ platform:iOS Simulator, id:E07261B8-E89D-42EC-91D2-B625A2ABD5B4, OS:10.1, name:iPad Air }<br>
{ platform:iOS Simulator, id:133AC706-9A8A-4B7A-8F7E-4EC123936B67, OS:10.2, name:iPad Air }<br>
{ platform:iOS Simulator, id:F7239337-DF32-4A27-ABC1-C9D826D543D5, OS:13.0, name:iPad Air (3rd generation) }<br>
{ platform:iOS Simulator, id:83906665-3620-4837-90C4-29853D2E683D, OS:10.0, name:iPad Air 2 }<br>
{ platform:iOS Simulator, id:0CBA8BE6-BD2B-43C1-97E7-5905FA930030, OS:10.1, name:iPad Air 2 }<br>
{ platform:iOS Simulator, id:DD2CC71A-F36C-4E59-B374-95FD1CEFAF25, OS:10.2, name:iPad Air 2 }<br>
{ platform:iOS Simulator, id:749E3FAB-948D-4EEE-8BEE-8EA848B189D5, OS:10.0, name:iPad Pro (9.7 inch) }<br>
{ platform:iOS Simulator, id:E4E95709-85C0-4214-9094-6567F607959F, OS:10.1, name:iPad Pro (9.7 inch) }<br>
{ platform:iOS Simulator, id:B6C5A6B2-12C9-4A9F-86B9-A286E34606D4, OS:10.2, name:iPad Pro (9.7 inch) }<br>
{ platform:iOS Simulator, id:2E10BF9F-F149-4E3D-AF05-833E7F602C38, OS:13.0, name:iPad Pro (9.7-inch) }<br>
{ platform:iOS Simulator, id:8D4D80CE-C4FF-4EE0-ADEE-C1E5A5E89294, OS:13.0, name:iPad Pro (11-inch) }<br>
{ platform:iOS Simulator, id:9820E433-6BBA-4909-8F1A-5D640B382CB8, OS:10.0, name:iPad Pro (12.9 inch) }<br>
{ platform:iOS Simulator, id:D51E2EB6-5C0C-47E2-A441-E4DB465EA51D, OS:10.1, name:iPad Pro (12.9 inch) }<br>
{ platform:iOS Simulator, id:2904B725-369E-422D-8CD2-7238AE298753, OS:10.2, name:iPad Pro (12.9 inch) }<br>
{ platform:iOS Simulator, id:C2503192-3AF2-4F9E-9E54-D6DFE62A1FAE, OS:13.0, name:iPad Pro (12.9-inch) (3rd generation) }<br>
{ platform:iOS Simulator, id:5983F321-AACC-4841-8FBC-70ACAE19E7B5, OS:10.0, name:iPhone 5 }<br>
{ platform:iOS Simulator, id:EC4382A1-7ADD-4F0A-B9C6-ADEB36F0D7E2, OS:10.1, name:iPhone 5 }<br>
{ platform:iOS Simulator, id:152DB888-156E-474F-97B8-A1E06746AED3, OS:10.2, name:iPhone 5 }<br>
{ platform:iOS Simulator, id:F3EA5819-7F51-49FC-A44B-9B48FDBE0A71, OS:10.0, name:iPhone 5s }<br>
{ platform:iOS Simulator, id:7BF64C3F-CF5B-4C33-819B-C7E8B2A7A4C2, OS:10.1, name:iPhone 5s }<br>
{ platform:iOS Simulator, id:ED122C35-A3D5-4980-B748-D5494AFD3582, OS:10.2, name:iPhone 5s }<br>
{ platform:iOS Simulator, id:1519FABF-1F9E-4921-B1A8-7ADE2CE9C29D, OS:10.0, name:iPhone 6 }<br>
{ platform:iOS Simulator, id:C3F1069A-5A98-476A-B9F1-174332BF77BE, OS:10.1, name:iPhone 6 }<br>
{ platform:iOS Simulator, id:20FA0158-31F3-4E0A-BEE5-7844528401F0, OS:10.2, name:iPhone 6 }<br>
{ platform:iOS Simulator, id:6B3740ED-5A8A-4710-BD87-ED808DFC0542, OS:10.0, name:iPhone 6 Plus }<br>
{ platform:iOS Simulator, id:E48D66D6-43D1-433C-A478-D96A3075A250, OS:10.1, name:iPhone 6 Plus }<br>
{ platform:iOS Simulator, id:2739E781-4175-4627-87AC-715CF8CBA832, OS:10.2, name:iPhone 6 Plus }<br>
{ platform:iOS Simulator, id:6972CDA3-E940-49C9-8D18-D87B4FB97E80, OS:10.0, name:iPhone 6s }<br>
{ platform:iOS Simulator, id:B353A0B7-A5BF-4F9D-984A-50F1D1612C65, OS:10.1, name:iPhone 6s }<br>
{ platform:iOS Simulator, id:A10119DA-328F-4BE0-818F-9B3A5448F83F, OS:10.2, name:iPhone 6s }<br>
{ platform:iOS Simulator, id:C3664329-11FD-46C3-8707-D3A02DCDC23C, OS:10.0, name:iPhone 6s Plus }<br>
{ platform:iOS Simulator, id:75F78647-E5B8-473B-8127-63A4C1E50969, OS:10.1, name:iPhone 6s Plus }<br>
{ platform:iOS Simulator, id:052381F4-7D60-4873-A67A-0B0B9C01A6FC, OS:10.2, name:iPhone 6s Plus }<br>
{ platform:iOS Simulator, id:9190CD6E-44EA-4B3D-B744-D6212AF1EB1E, OS:10.1, name:iPhone 7 }<br>
{ platform:iOS Simulator, id:01384E58-D752-4513-85E1-04FC9028B057, OS:10.2, name:iPhone 7 }<br>
{ platform:iOS Simulator, id:C4362690-AF3D-478F-B109-C0FF0BB74B1F, OS:10.1, name:iPhone 7 Plus }<br>
{ platform:iOS Simulator, id:FE62E13A-0E14-4D1C-A34E-48EFBA26367C, OS:10.2, name:iPhone 7 Plus }<br>
{ platform:iOS Simulator, id:C781302B-9B94-48CD-AFD1-B9092ED6852F, OS:13.0, name:iPhone 8 }<br>
{ platform:iOS Simulator, id:4FE93C5C-A455-41B0-902E-3951D6E0156D, OS:13.0, name:iPhone 8 Plus }<br>
{ platform:iOS Simulator, id:A85A40DC-270E-47AF-9854-4F33F765E356, OS:13.0, name:iPhone 11 }<br>
{ platform:iOS Simulator, id:DBF2165D-0798-4276-A1D5-4EF08EF178A1, OS:13.0, name:iPhone 11 Pro }<br>
{ platform:iOS Simulator, id:B5FCD37E-A6EA-40FF-93FA-8385E3171F59, OS:13.0, name:iPhone 11 Pro Max }<br>
{ platform:iOS Simulator, id:A21AD230-94DE-4BE0-B1F7-8296E6CA2403, OS:10.0, name:iPhone SE }<br>
{ platform:iOS Simulator, id:FE97D195-3ABA-4280-9772-63BA955E1E72, OS:10.1, name:iPhone SE }<br>
{ platform:iOS Simulator, id:004D7054-2FD1-4043-9DC3-5C5E196A5158, OS:10.2, name:iPhone SE }</p>
<p>Where are the iOS 12 and iOS 11 simulators? The 10.3.1 simulator? Where are the iPhone X series devices?</p>
<p>P.S. How is it that these forums don’t support Markdown?</p>

**Comment by chrispat** [comment, marked as answer]
<p>Please open feature requests and feedback for the virtual environments here <a href="https://github.com/actions/virtual-environments" rel="nofollow noopener">https://github.com/actions/virtual-environments</a>.</p>
